### PR TITLE
Add support for absolute file paths in all relevant scan options

### PR DESCRIPTION
**Pull request by jporzuceksfx**
Resolves https://github.com/zaproxy/zaproxy/issues/6547

### CODE REVIEW

**Comment by psiinon** [comment]
Thanks!
Could you add a like to the changelog: https://github.com/zaproxy/zaproxy/blob/main/docker/CHANGELOG.md

**Review by psiinon — APPROVED** [review]

File: docker/CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,9 @@
 # Changelog
 All notable changes to the docker containers will be documented in this file.
 
+### 2022-03-28
+ - Add support for absolute filepath in API Scan target option.
```
Comment (kingthorin) [inline docker/CHANGELOG.md:5]:
```suggestion
- Add support for absolute filepath in API Scan target option.
```

Comment (jporzuceksfx) [reply docker/CHANGELOG.md:5]: @kingthorin Looks like a space before a dash is in harmony with other entries in changelog file.

**Comment by thc202** [comment]
IMO other file based arguments should behave the same, for consistency.

**Comment by jporzuceksfx** [comment]
@thc202 I went ahead and made those changes for all applicable options in all scan types.

**Comment by thc202** [comment]
Thank you, will check it today.

File: docker/CHANGELOG.md [hunk 2]
```diff
@@ -1,6 +1,9 @@
 # Changelog
 All notable changes to the docker containers will be documented in this file.
 
+### 2022-03-31
+ - Add support for absolute filepath in scan target options which require filename parameter.
```
Comment (thc202) [inline docker/CHANGELOG.md:5]: Should say "in all options" now?

File: docker/zap-api-scan.py [hunk 3]
```diff
@@ -147,7 +147,7 @@ def main(argv):
     delay = 0
     timeout = 0
     ignore_warn = False
-    hook_file = None
+    hook_file = ''
```
Comment (thc202) [inline docker/zap-api-scan.py:150]: Why this change?

File: docker/CHANGELOG.md [hunk 4]
```diff
@@ -1,6 +1,9 @@
 # Changelog
 All notable changes to the docker containers will be documented in this file.
 
+### 2022-03-31
```
Comment (thc202) [inline docker/CHANGELOG.md:4]: Needs update.

File: docker/zap-baseline.py [hunk 5]
```diff
@@ -341,8 +342,8 @@ def main(argv):
 
             # Generate the yaml file
             home_dir = str(Path.home())
-            yaml_file = home_dir + '/zap.yaml'
-            summary_file = home_dir + '/zap_out.json'
+            yaml_file = os.path.join(home_dir,'zap.yaml')
```
Comment (thc202) [inline docker/zap-baseline.py:345]: Add space after the comma.

**Comment by thc202** [comment]
Could you fix up the commits?

**Comment by kingthorin** [comment]
Could you remove the unrelated commits please?

**Comment by jporzuceksfx** [comment]
@kingthorin Commits squeezed into one

**Comment by kingthorin** [comment]
Thanks that looks better.

**Review by kingthorin — APPROVED** [review]

**Comment by thc202** [comment]
Will do.

**Comment by thc202** [comment]
Updated the changelog to match the current date and tweak the description of the change.

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!
